tests/test_project_structure.py

```
@@ -0,0 +1,303 @@
+    def test_cleanup_removes_temp_files(self):
+        """Test that temp files are removed from output."""
+        with tempfile.TemporaryDirectory() as tmpdir:
+            init_project(tmpdir)
+
+            # Create temp file in output
+            output_dir = Path(tmpdir) / "output" / "test_pipeline"
+            output_dir.mkdir(parents=True)
+            temp_file = output_dir / "step_1_output.txt"
+            temp_file.write_text("temp data")
+
+            count, deleted = cleanup_temp_files(tmpdir)
```

Ruff (0.14.14)

[warning] Unpacked variable `deleted` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)